M2/Macaulay2/packages/MultiprojectiveVarieties.m2

```diff
@@ -96,21 +98,21 @@ projectiveVariety Ideal := o -> I -> (
 projectiveVariety Ring := o -> R -> (
     if R#?"multiprojectiveVariety" then return R#"multiprojectiveVariety";
     I := ideal R;
-    if o.Saturate then if not isMultisaturated I then error "the ideal is not multi-saturated";
+    if not isPolynomialRing ambient R then error "expected the ambient ring to be polynomial";
+    if o.Saturate or (I#cache)#?"isMultisaturated" then if not isMultisaturated I then error "the ideal is not multi-saturated";
     if (I#cache)#?"multiprojectiveVariety" then (
         X := (I#cache)#"multiprojectiveVariety";
         if X#"ringVariety" === null then X#"ringVariety" = R;
         if X#"ringVariety" === R then return (R#"multiprojectiveVariety" = X);
     );
-    if not isPolynomialRing ambient R then error "expected the ambient ring to be polynomial";
     Y := projectiveVariety(I,MinimalGenerators=>false,Saturate=>false);
     if Y#"ringVariety" =!= null then error "internal error encountered: double assignment for ring of projective variety";
     Y#"ringVariety" = R;
     R#"multiprojectiveVariety" = Y
 );
 
 isMultisaturated = I -> (
```

Comment:
I think this would be much simpler:
```m2
isMultisaturated = (cacheValue symbol isMultisaturated) (I -> I == multisaturate I)
```

Reply:
The result would be stored under `I.cache.isMultisaturated`.

Reply:
Yes, this shortens the code a bit.